src/set_theory/cardinal/basic.lean

```diff
@@ -564,22 +565,27 @@ cardinal.wf.conditionally_complete_linear_order_with_bot 0 $ le_antisymm (cardin
 
 instance wo : @is_well_order cardinal.{u} (<) := ⟨cardinal.wf⟩
 
+/-- The set in the definition of `cardinal.succ` is nonempty. -/
+theorem succ_nonempty (c : cardinal) : {c' : cardinal | c < c'}.nonempty := ⟨_, cantor c⟩
+
 /-- The successor cardinal - the smallest cardinal greater than
   `c`. This is not the same as `c + 1` except in the case of finite `c`. -/
 def succ (c : cardinal) : cardinal :=
 Inf {c' | c < c'}
 
-theorem succ_nonempty (c : cardinal) : {c' : cardinal | c < c'}.nonempty :=
-⟨_, cantor _⟩
-
-theorem lt_succ_self (c : cardinal) : c < succ c :=
+theorem lt_succ (c : cardinal) : c < succ c :=
 Inf_mem (succ_nonempty c)
 
-theorem succ_le {a b : cardinal} : succ a ≤ b ↔ a < b :=
-⟨lt_of_lt_of_le (lt_succ_self _), λ h, cInf_le' h⟩
+theorem succ_le_iff {a b : cardinal} : succ a ≤ b ↔ a < b :=
+⟨(lt_succ a).trans_le, λ h, cInf_le' h⟩
+
+instance : succ_order cardinal :=
+succ_order.of_succ_le_iff succ (λ a b, succ_le_iff)
```

Comment:
I'd be inclined to swap these and implement `succ_le_iff` as `order.succ_le_iff`; that makes it more obvious that all the lemmas can be replaced with their order versions

Reply:
I need `succ_le_iff` in order to define the instance though, so I don't think this is any better.

At the request of Yaël, I'll rework this PR so that `ordinal.succ` is replaced with `order.succ`.

Reply:
I think that would make more sense to do once `cardinal.succ` is gone.

Reply:
Otherwise, we'd have to prove `succ_le_iff` once to build the instance, and then we'd prove the exact same statement with the exact same name via `order.succ_le_iff`. Doesn't make much sense to me.